### PR TITLE
Add type annotations part 1

### PR DESCRIPTION
**Pull request by benbariteau**
This annotates the `kafka_utils.kafka_check` module. It doesn't have any real effect right now, but this repo is large so annotating in chunks makes sense.

I also migrated from retrying to tenacity, since retrying is a dead package and tenacity has type annotations.

### CODE REVIEW

File: kafka_utils/kafka_check/status_code.py [hunk 1]
```diff
@@ -26,14 +31,19 @@
 }
 
 
-def prepare_terminate_message(string):
+class TerminateMessageDict(TypedDict):
+    message: str
+    raw: str
+
+
+def prepare_terminate_message(string: str) -> TerminateMessageDict:
```
Comment (JayH5) [inline kafka_utils/kafka_check/status_code.py:39]: It seems like every time we use this function we cast to a `dict[str, Any]` anyway? What's the general approach you are using to decide when to do stricter typing (e.g. `TypedDict` stuff) vs simpler/less code (e.g. just mark the return type as `-> dict` or something)? Or is it all just generated by some tool?

**Review by JayH5 — APPROVED** [review]

File: setup.py [hunk 2]
```diff
@@ -58,7 +58,8 @@
         "requests-futures>0.9.0",
         "paramiko>1.8.0,<3.0.0",
         "requests<3.0.0",
-        "retrying",
+        "tenacity",
+        "typing-extensions>=3.7.4",
```
Comment (JayH5) [inline setup.py:62]: Is the `py.typed` stuff coming in part 2?

Comment (benbariteau) [reply setup.py:62]: Probably a much later part. My guess is 8-10 PRs for typing this.

File: kafka_utils/kafka_check/status_code.py (hunk 1 above)
Comment (benbariteau) [reply kafka_utils/kafka_check/status_code.py:39]: That's a good question and I don't have a perfect answer. The reason I used `TerminateMessageDict` in this case is that it has a concrete type. However, the usages of it were more general. Maybe it's not worth it in practice, but I prefer stricter types like this where possible. In theory it makes refactoring this to code to be better designed from a type safety perspective easier.

**Pull request merged**